core/dbt/config/selectors.py

```diff
@@ -118,6 +119,25 @@ def selector_config_from_data(
     return selectors
 
 
+def validate_selector_default(selector_file: SelectorFile) -> None:
+    """Check if a selector.yml file has more than 1 default key set to true"""
+    default_set: bool = False
+    default_selector_name: Union[str, None] = None
+
+    for selector in selector_file.selectors:
+        if selector.default is True and default_set is False:
+            default_set = True
+            default_selector_name = selector.name
+            continue
+        if selector.default is True and default_set is True:
+            raise DbtSelectorsError(
+                "Error when parsing the selector file. "
+                "Default key is set to True multiple times.\n"
+                "Default with value set to True found in "
+                f"{default_selector_name} and {selector.name}"
```

Comment:
Nice! Let's make it clear that `default: true` is found in multiple different selectors. Perhaps something like:
```python
                f"Found multiple selectors with `default: true`:"
                f"{default_selector_name} and {selector.name}"
```